src/box/msgpack.c

```diff
@@ -89,9 +89,32 @@ msgpack_snprint_ext(char *buf, int size, const char **data, int depth)
 	}
 }
 
+/** Our handler to validate MP_EXT contents. */
+static int
+msgpack_check_ext_data(int8_t type, const char *data, uint32_t len)
+{
+	switch (type) {
+	case MP_DECIMAL:
+		return mp_validate_decimal(data, len);
+	case MP_UUID:
+		return mp_validate_uuid(data, len);
+	case MP_DATETIME:
+		return mp_validate_datetime(data, len);
+	case MP_ERROR:
+		return mp_validate_error(data, len);
+	case MP_COMPRESSION:
+		return mp_validate_compression(data, len);
+	case MP_INTERVAL:
+		return mp_validate_interval(data, len);
+	default:
+		return mp_check_ext_data_default(type, data, len);
```

Comment:
`mp_check_ext_data_default` always returns 0. Shouldn't we return an error in case of an unknown extension?

Also, I don't think we need to handle `MP_COMPRESSION` here at all - it's an internal type used solely for encoding tuples in memtx so end users should never see it nor use it.